web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/TextSearchFilterServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+        this.calculatePatientAgeService = calculatePatientAgeService;
+        this.calculatePatientGenderService = calculatePatientGenderService;
+        this.diagnosisChapterService = diagnosisChapterService;
+    }
+
+    @Override
+    public List<SjukfallEnhet> filterList(List<SjukfallEnhet> sickLeaves, String textSearch) {
+        if (textSearch == null || textSearch.isEmpty()) {
+            return sickLeaves;
+        }
+        return filterOnTextSearch(sickLeaves, textSearch);
+    }
+
+    @Override
+    public boolean filter(SjukfallEnhet sickLeave, String textSearch) {
+        if (textSearch == null || textSearch.isEmpty()) {
+            return true;
+        }
+        return textSearchMatchesAnyField(sickLeave, textSearch);
+    }
+
+    private List<SjukfallEnhet> filterOnTextSearch(List<SjukfallEnhet> sickLeaves, String textSearch) {
+        return sickLeaves.stream()
+            .filter(sickLeave -> textSearchMatchesAnyField(sickLeave, textSearch))
+            .collect(Collectors.toList());
+    }
+
+    private boolean textSearchMatchesAnyField(SjukfallEnhet sickLeave, String textSearch) {
+        return searchMatchesAge(sickLeave, textSearch) || searchMatchesPatientName(sickLeave, textSearch) || searchMatchesPatientGender(
+            sickLeave, textSearch) || searchMatchesDiagnosisCode(sickLeave, textSearch) || searchMatchesSickLeavePeriod(sickLeave,
+            textSearch) || searchMatchesLength(sickLeave, textSearch) || searchMatchesNumberOfCertificates(sickLeave, textSearch)
+            || searchMatchesActiveDegree(sickLeave, textSearch) || searchMatchesDoctorName(sickLeave, textSearch) || searchMatchesPatientId(
+            sickLeave, textSearch);
+    }
+
+    private boolean searchMatchesPatientId(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getPatient().getId(), textSearch);
+    }
+
+    private boolean searchMatchesDoctorName(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getLakare().getNamn(), textSearch);
+    }
+
+    private boolean searchMatchesActiveDegree(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getAktivGrad() + "%", textSearch);
+    }
+
+    private boolean searchMatchesNumberOfCertificates(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(String.valueOf(sickLeave.getIntygLista().size()), textSearch);
+    }
+
+    private boolean searchMatchesLength(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(String.valueOf(sickLeave.getDagar()), textSearch);
+    }
+
+    private boolean searchMatchesSickLeavePeriod(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getStart().toString(), textSearch) || matches(sickLeave.getSlut().toString(), textSearch);
+    }
+
+    private boolean searchMatchesDiagnosisCode(SjukfallEnhet sickLeave, String textSearch) {
```

Comment:
As discussed no need to check on chapter.. instead just check diagnose code and name (for diagnosKod and biDiagnoser)